pkg/component/delete/delete.go

```diff
@@ -87,54 +87,60 @@ func references(list []unstructured.Unstructured, ownerRef metav1.OwnerReference
 }
 
 // ListResourcesToDeleteFromDevfile parses all the devfile components and returns a list of resources that are present on the cluster and can be deleted
-func (do DeleteComponentClient) ListResourcesToDeleteFromDevfile(devfileObj parser.DevfileObj, appName string) (isInnerLoopDeployed bool, resources []unstructured.Unstructured, err error) {
-	// Inner Loop
-	// Fetch the deployment of the devfile component
-	componentName := devfileObj.GetMetadataName()
-	var deploymentName string
-	deploymentName, err = util.NamespaceKubernetesObject(componentName, appName)
-	if err != nil {
-		return isInnerLoopDeployed, resources, fmt.Errorf("failed to get the resource %q name for component %q; cause: %w", kclient.DeploymentKind, deploymentName, err)
-	}
+func (do DeleteComponentClient) ListResourcesToDeleteFromDevfile(devfileObj parser.DevfileObj, appName string, mode string) (isInnerLoopDeployed bool, resources []unstructured.Unstructured, err error) {
+
+	if mode == odolabels.ComponentDevMode || mode == odolabels.ComponentAnyMode {
+		// Inner Loop
+		// Fetch the deployment of the devfile component
+		componentName := devfileObj.GetMetadataName()
+		var deploymentName string
+		deploymentName, err = util.NamespaceKubernetesObject(componentName, appName)
+		if err != nil {
+			return isInnerLoopDeployed, resources, fmt.Errorf("failed to get the resource %q name for component %q; cause: %w", kclient.DeploymentKind, deploymentName, err)
```

Comment:
I think this should be `componentName` and not `deploymentName`. Also, if there was an error, the normal expectation is for `deploymentName` variable to not contain any value, no?
```suggestion
			return isInnerLoopDeployed, resources, fmt.Errorf("failed to get the resource %q name for component %q; cause: %w", kclient.DeploymentKind, componentName, err)
```